frontend/src/stores/accounts.ts

```diff
@@ -2,67 +2,187 @@ import { defineStore } from 'pinia';
 import { computed, ref } from 'vue';
 import type { Address } from '@/types';
 import { createAccount, generatePrivateKey } from 'genlayer-js';
-import type { Account } from 'genlayer-js/types';
 import { useShortAddress } from '@/hooks';
 
+export interface AccountInfo {
+  type: 'local' | 'metamask';
+  address: Address;
+  privateKey?: Address; // Only for local accounts
+}
+
 export const useAccountsStore = defineStore('accountsStore', () => {
-  const key = localStorage.getItem('accountsStore.currentPrivateKey');
-  const currentPrivateKey = ref<Address | null>(key ? (key as Address) : null);
-  const currentUserAddress = computed(() =>
-    currentPrivateKey.value
-      ? createAccount(currentPrivateKey.value).address
-      : '',
-  );
   const { shorten } = useShortAddress();
 
-  const privateKeys = ref<Address[]>(
-    localStorage.getItem('accountsStore.privateKeys')
-      ? ((localStorage.getItem('accountsStore.privateKeys') || '').split(
-          ',',
-        ) as Address[])
-      : [],
+  // Store all accounts (both local and MetaMask)
+  const accounts = ref<AccountInfo[]>([]);
+  const selectedAccount = ref<AccountInfo | null>(null);
+
+  // Migrate from old storage to new storage
+  const storedKeys = localStorage.getItem('accountsStore.privateKeys');
+  if (storedKeys) {
+    const privateKeys = storedKeys.split(',') as Address[];
+    accounts.value = privateKeys.map(createAccount);
+    localStorage.removeItem('accountsStore.privateKeys');
+    localStorage.removeItem('accountsStore.currentPrivateKey');
+    localStorage.removeItem('accountsStore.accounts');
+    _initAccountsLocalStorage();
+  }
+
+  // Initialize accounts from localStorage
+  const storedAccounts = JSON.parse(
+    localStorage.getItem('accountsStore.accounts') || '[]',
+  );
+  if (storedAccounts.length === 0) {
+    generateNewAccount();
+    _initAccountsLocalStorage();
+  } else {
+    accounts.value = storedAccounts;
+  }
+
+  // Initialize selected account from localStorage
+  const storedSelectedAccount = JSON.parse(
+    localStorage.getItem('accountsStore.currentAccount') || 'null',
   );
+  setCurrentAccount(
+    storedSelectedAccount ? storedSelectedAccount : accounts.value[0],
+  );
+
+  function _initAccountsLocalStorage() {
+    localStorage.setItem(
+      'accountsStore.accounts',
+      JSON.stringify(accounts.value),
+    );
+    localStorage.setItem(
+      'accountsStore.currentAccount',
+      JSON.stringify(accounts.value[0]),
+    );
+  }
+
+  async function fetchMetaMaskAccount() {
+    if (window.ethereum) {
```

Comment:
Here I think we should handle the case were !window.ethereum:

```suggestion
    if (!window.ethereum) {
      notify({  
        title: 'MetaMask is not installed',
        type: 'error',
      });
      return;
    }
```

Reply:
Will need to add

`import { notify } from '@kyvg/vue3-notification';`

Reply:
it would be fine if metamask is not installed in the browser. Users could still use the current local accounts